sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -1205,7 +1205,7 @@ class UDFSuite extends QueryTest with SharedSparkSession {
         dt
       )
       checkError(
-        intercept[AnalysisException](spark.range(1).select(f())),
+        intercept[AnalysisException](spark.range(1).select(f()).encoder),
```

Comment:
I think this is fine, we will remove this error eventually after completely support char/varchar (it's in progress)